pkg/apb/types.go

```diff
@@ -34,18 +34,30 @@ type SpecManifest map[string]*Spec
 
 // ParameterDescriptor - a parameter to be used by the service catalog to get data.
 type ParameterDescriptor struct {
-	Name         string      `json:"name"`
-	Title        string      `json:"title"`
-	Type         string      `json:"type"`
-	Description  string      `json:"description,omitempty"`
-	Default      interface{} `json:"default,omitempty"`
-	Maxlength    int         `json:"maxlength,omitempty"`
-	Pattern      string      `json:"pattern,omitempty"`
-	Enum         []string    `json:"enum,omitempty"`
-	Required     bool        `json:"required"`
-	Updatable    bool        `json:"updatable"`
-	DisplayType  string      `json:"display_type,omitempty" yaml:"display_type,omitempty"`
-	DisplayGroup string      `json:"display_group,omitempty" yaml:"display_group,omitempty"`
+	Name        string      `json:"name"`
+	Title       string      `json:"title"`
+	Type        string      `json:"type"`
+	Description string      `json:"description,omitempty"`
+	Default     interface{} `json:"default,omitempty"`
+
+	// string validators
+	DeprecatedMaxlength int    `json:"maxlength,omitempty" yaml:"maxlength,omitempty"` // backwards compatibility
+	MaxLength           int    `json:"maxLength,omitempty" yaml:"max_length,omitempty"`
+	MinLength           int    `json:"minLength,omitempty" yaml:"min_length,omitempty"`
+	Pattern             string `json:"pattern,omitempty"`
+
+	// number validators
+	MultipleOf       float64     `json:"multipleOf,omitempty" yaml:"multiple_of,omitempty"`
+	Maximum          interface{} `json:"maximum,omitempty"`
```

Comment:
I don't like the loss of type validation if someone is creating a ParameterDescription by hand. If we can make this not exported I would be much less worried about it. 

This is another way around this. Using a type declaration for float64 and dereferencing the pointer when using that value. Example: 

```go
type nilableNumber float64

type ParameterDescription struct {
.....
	Maximum          *nilableNumber `json:"maximum,omitempty"`
.....
}
```

https://play.golang.org/p/qpeFB6b8GXp